### PR TITLE
Include enrollment block when WAZUH_REGISTRATION_PASSWORD variable is defined

### PR DESCRIPTION
**Pull request by okynos**
|Related issue|
|---|
| https://github.com/wazuh/wazuh-packages/issues/1565 |

## Description
Hello team,

We have included a modification to include enrollment block in addition with `authorization_pass_path` inside to the `ossec.conf` and `enrollment` block.

## Configuration options
None

## Logs/Alerts example
None

## Tests
Tested package building and auto deploy installation with `WAZUH_REGISTRATION_PASSWORD` and `WAZUH_MANAGER` variables.
Tested with Windows and Linux agents

Command Linux:
```
WAZUH_MANAGER=172.17.0.2 WAZUH_REGISTRATION_PASSWORD='superSecret' WAZUH_REGISTRATION_SERVER=172.17.0.2 apt -y install /root/wazuh-agent_4.4.0-0.rp_amd64.deb
```

Conf Linux:
```
    <enrollment>
      <enabled>yes</enabled>
      <manager_address>172.17.0.2</manager_address>
      <authorization_pass_path>/var/ossec/etc/authd.pass</authorization_pass_path>
    </enrollment>
```

---

Command Windows:
```
./wazuh-agent-4.4.0-0.rp.msi /q WAZUH_MANAGER="172.17.0.2" WAZUH_REGISTRATION_PASSWORD="superSecret"
```

Conf Windows:
```
<enrollment>
  <enabled>yes</enabled>
  <authorization_pass_path>authd.pass</authorization_pass_path>
</enrollment>
```

### CODE REVIEW

File: src/init/register_configure_agent.sh [hunk 1]
```diff
@@ -13,6 +13,7 @@ INSTALLDIR=${1}
 CONF_FILE="${INSTALLDIR}/etc/ossec.conf"
 TMP_ENROLLMENT="${INSTALLDIR}/tmp/enrollment-configuration"
 TMP_SERVER="${INSTALLDIR}/tmp/server-configuration"
+WAZUH_REGISTRATION_PASSWORD_PATH="${INSTALLDIR}/etc/authd.pass"
```
Comment (DFolchA) [inline src/init/register_configure_agent.sh:16]: I would rather use the default value ` /etc/authd.pass`, as indicated in the documentation:
https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/client.html#authorization-pass-path

**Review by DFolchA — CHANGES_REQUESTED** [review]

File: src/init/register_configure_agent.sh (hunk 1 above)
Comment (okynos) [reply src/init/register_configure_agent.sh]: 2ecd9993ec

**Review by DFolchA — APPROVED** [review]
LGTM

**Review by alberpilot — APPROVED** [review]
LGTM

**Review by chemamartinez — APPROVED** [review]
LGTM!

**Review by alberpilot — APPROVED** [review]
LGTM

**Comment by jmv74211** [comment]
## QA review

- **Type**: Manual testing.
- **Status**: Approved  🟢 
- **Testing issue**: https://github.com/wazuh/wazuh-qa/issues/3124
- **Comments**: Nothing to highlight. Everything seems to work properly.

**Review by vikman90 — APPROVED** [review]
## Test on Ubuntu 22.04

### Command

```sh
wget https://packages-dev.wazuh.com/warehouse/pullrequests/4.4/deb/var/wazuh-agent_4.4.0-0.commitef68bf4_amd64.deb
WAZUH_MANAGER=172.17.18.239 WAZUH_REGISTRATION_PASSWORD='superSecret' WAZUH_REGISTRATION_SERVER=172.17.18.239 apt -y install
```

<details><summary>Output</summary>

```
Reading package lists... Done
Building dependency tree
Reading state information... Done
Note, selecting 'wazuh-agent' instead of './wazuh-agent_4.4.0-0.commitef68bf4_amd64.deb'
The following NEW packages will be installed:
  wazuh-agent
0 upgraded, 1 newly installed, 0 to remove and 13 not upgraded.
Need to get 0 B/8,731 kB of archives.
After this operation, 28.1 MB of additional disk space will be used.
Get:1 /root/wazuh-agent_4.4.0-0.commitef68bf4_amd64.deb wazuh-agent amd64 4.4.0-0.commitef68bf4 [8,731 kB]
Preconfiguring packages ...
Selecting previously unselected package wazuh-agent.
(Reading database ... 147173 files and directories currently installed.)
Preparing to unpack .../wazuh-agent_4.4.0-0.commitef68bf4_amd64.deb ...
Unpacking wazuh-agent (4.4.0-0.commitef68bf4) ...
Setting up wazuh-agent (4.4.0-0.commitef68bf4) ...
Processing triggers for systemd (245.4-4ubuntu3.16) ...
N: Download is performed unsandboxed as root as file '/root/wazuh-agent_4.4.0-0.commitef68bf4_amd64.deb' couldn't be accessed by user '_apt'. - pkgAcquire::Run (13: Permission denied)
```

</details>

### Agent configuration

```xml
<client>
  <server>
    <address>172.17.18.239</address>
    <port>1514</port>
    <protocol>tcp</protocol>
  </server>
  <config-profile>ubuntu, ubuntu20, ubuntu20.04</config-profile>
  <notify_time>10</notify_time>
  <time-reconnect>60</time-reconnect>
  <auto_restart>yes</auto_restart>
  <crypto_method>aes</crypto_method>
  <enrollment>
    <enabled>yes</enabled>
    <manager_address>172.17.18.239</manager_address>
    <authorization_pass_path>etc/authd.pass</authorization_pass_path>
  </enrollment>
</client>
```

### Enrollment password

```sh
cat /var/ossec/etc/authd.pass
```

```
superSecret
```

### Logs

```
2022/07/28 19:30:03 wazuh-agentd: INFO: Requesting a key from server: 172.17.18.239
2022/07/28 19:30:03 wazuh-agentd: INFO: Using password specified on file: etc/authd.pass
2022/07/28 19:30:03 wazuh-agentd: INFO: Using agent name as: focal
2022/07/28 19:30:03 wazuh-agentd: INFO: Waiting for server reply
2022/07/28 19:30:03 wazuh-agentd: INFO: Valid key received
```

**Pull request merged**